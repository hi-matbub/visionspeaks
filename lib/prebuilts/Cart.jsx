import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-bootstrap/Modal';
import CartPlaceholder from '../components/CartPlaceholder';
import CartModal from './CartModal';

const Cart = ({ cart, handleUpdatedCartInState, currentBill }) => {
  const [isCartEmpty, setCartToEmpty] = useState(true);

  // modal funcs
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  useEffect(() => {
    // eslint-disable-next-line no-unused-expressions
    cart ? setCartToEmpty(false) : setCartToEmpty(true);
  }, [cart]);

  const handleEmptyCartNotice = () => setCartToEmpty(true);

  return (
    <>
      <CartPlaceholder handleShow={handleShow} isCartEmpty={isCartEmpty} />

      <Modal show={show} onHide={handleClose}>
        {isCartEmpty ? (
          <h2 className="m-4">Nothing in cart</h2>
        ) : (
          <CartModal
            cart={cart}
            currentBill={currentBill}
            handleClose={handleClose}
            handleEmptyCartNotice={handleEmptyCartNotice}
            handleUpdatedCartInState={(updatedCart) =>
              handleUpdatedCartInState(updatedCart)
            }
          />
        )}
      </Modal>
    </>
  );
};

Cart.propTypes = {
  cart: PropTypes.shape({}),
  data: PropTypes.shape({}).isRequired,
  handleUpdatedCartInState: PropTypes.func.isRequired,
  currentBill: PropTypes.number,
};

Cart.defaultProps = {
  cart: {},
  currentBill: 0,
};

export default Cart;
