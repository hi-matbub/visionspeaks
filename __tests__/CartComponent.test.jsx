import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Cart from '../lib/prebuilts/Cart';

const product = {
  name: 'product',
  description: 'this is an awesome product you should buy 2',
  price: '$2.87',
  id: 'abc_123',
  images: [
    '/path_01.png',
    '/path_02.png',
    '/path_03.png',
  ],
  qty: 12,
};

const handleUpdatedCartInState = () => {};

it('CheckoutError component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Cart 
      cart={{product}} 
      handleUpdatedCartInState={handleUpdatedCartInState} 
    />
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it('CheckoutError component renders unchanged', () => {
  const tree = renderer.create(
    <Cart
      cart={{product}} 
      handleUpdatedCartInState={handleUpdatedCartInState} 
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
