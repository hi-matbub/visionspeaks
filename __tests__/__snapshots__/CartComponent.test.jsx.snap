// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutError component renders unchanged 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "background": "none",
      "border": "0",
    }
  }
  type="button"
>
  <div
    className="bg-warning scale-in-center"
    style={
      Object {
        "borderRadius": "50%",
        "height": "15px",
        "position": "absolute",
        "right": "14px",
        "top": "28px",
        "width": "15px",
      }
    }
  />
  <img
    alt="Checkout"
    src="/cart.png"
    style={
      Object {
        "cursor": "pointer",
        "height": "20px",
      }
    }
  />
</button>
`;
